gsdx windows: update gui to add a HPO v2 checkbox

gregory38 opened the pull request:
Squash of PR #1758 

Need someone to check the result. I didn't understand why `IDC_OFFSETHACK_NEW` was added so I kept only `IDC_OFFSETHACK_TEXT` (that being said, it is maybe useless too)

ssakash commented on plugins/GSdx/resource.h:
```diff
@@ -140,7 +141,7 @@
 #define _APS_NEXT_RESOURCE_VALUE        10013
 #define _APS_NEXT_COMMAND_VALUE         32771
-#define _APS_NEXT_CONTROL_VALUE         2106
+#define _APS_NEXT_CONTROL_VALUE         2107
```
Duplicate value? This macro should have a value of 2121.

ssakash commented on plugins/GSdx/GSSettingsDlg.cpp:
```diff
@@ -723,7 +722,12 @@ void GSHacksDlg::OnInit()
+	{
+		hpo_combobox.erase(hpo_combobox.begin() + 2, hpo_combobox.begin() + 4); //TODO: Enums would be more nicer than magic numbers.
+	}
+	ComboBoxInit(IDC_OFFSETHACK, hpo_combobox,theApp.GetConfigI("UserHacks_HalfPixelOffset");
```
Function missing closing paranthesis.

ssakash commented on plugins/GSdx/GSSettingsDlg.cpp:
```diff
@@ -723,7 +722,12 @@ void GSHacksDlg::OnInit()
+	std::vector<GSSetting> hpo_combobox = theApp.m_gs_offset_hack;
+	if (!ogl)
+	{
+		hpo_combobox.erase(hpo_combobox.begin() + 2, hpo_combobox.begin() + 4); //TODO: Enums would be more nicer than magic numbers.
```
Well, best to remove the comment here if there are no plans to add enumerations.

ssakash commented:
> Need someone to check the result. I didn't understand why IDC_OFFSETHACK_NEW was added so I kept only IDC_OFFSETHACK_TEXT (that being said, it is maybe useless too)

The define is needed, it's used to the gray out the respective GUI element when the user has selected a native resolution.

ssakash commented:
By the way, since you've made a PR on your own, maybe you could also add the necessary Linux/INI variable changes in this PR and merge this in a single go?

gregory38 commented:
@ssakash I will add the linux bit later. First I want to have be sure woj work is correct. I updated based on your comment. I added back the `_NEW` define.

ssakash commented:
> I updated based on your comment. I added back the _NEW define.

Ahh, I'm terribly sorry, I think I got it confused. That macro isn't needed, only the ``IDC_OFFSETHACK_TEXT`` define is needed. (``IDC_OFFSETHACK_TEXT`` is used for graying out the element)

gregory38 commented:
@ssakash you're lucky I didn't close my text editor :p I revert it. Just increase the final  value to 2121

woj1993 commented:
@gregory38 Thanks. Without help I probably would be still fixing it.

ssakash commented:
> You need to delete line 751 or 757 because is it doubled. Everything else seems ok.

Nice catch, though make sure to do the redundant function removal in a way that ``EnableWindow()`` for ``IDC_OFFSETHACK_TEXT`` and ``IDC_OFFSETHACK`` are close together.

FlatOutPS2 commented:
Seems like it's done now. I've tested it and it looks good. Ready to merge?

woj1993 commented:
@FlatOutPS2 I don't find any errors.
P.S. What I need to do with my pull request?

gregory38 commented:
I added Linux change. It misses @ssakash approval.

@woj1993 I will close both PR. However you might need to recreate a fresh GIT for future contribution. I know it wasn't easy, thanks you for your perseverance.

turtleli commented on plugins/GSdx/GSSetting.cpp:
```diff
@@ -55,7 +55,10 @@ const char* dialog_message(int ID, bool* updateText) {
+			return "Might fix some misaligned fog, bloom, or blend effect.\n"
+				"The preferred option is Normal (Vertex) as it is most likely to resolve misalignment issues.\n"
+				"The special cases are only useful in couple of games like Captain Tsubasa.\n"
+				"\n Special cases are disabled from code side for now.";
```
Space between `\n` and Special should be removed.

FlatOutPS2 replied:
The line can be removed instead. ;)

gregory38 replied:
Ok latest commit ammended :+1:

turtleli commented on plugins/GSdx/GSSetting.cpp:
```diff
@@ -55,7 +55,9 @@ const char* dialog_message(int ID, bool* updateText) {
-			return "Might fix some misaligned fog, bloom, or blend effect.";
+			return "Might fix some misaligned fog, bloom, or blend effect.\n"
+				"The preferred option is Normal (Vertex) as it is most likely to resolve misalignment issues.\n"
+				"The special cases are only useful in couple of games like Captain Tsubasa.\n";
```
`\n` should be removed from the end of the line. Also `a` should be inserted before `couple`.

gregory38 replied:
Done on first commit. Thank

woj1993 commented:
@gregory38 When Users hack HPO will have value 2 what will happen?

ssakash commented:
> @gregory38 When Users hack HPO will have value 2 what will happen?

The false branch of ``if (UserHacks_HPO == 3)`` will be entered, that's where the necessary functional code for ``Special (Aggressive)`` is present.

woj1993 commented:
@ssakash Ok. Sorry I didn't view full files.
